api/server/router/network/network_routes.go

```diff
@@ -79,8 +80,16 @@ func (n *networkRouter) postNetworkCreate(ctx context.Context, w http.ResponseWr
 		return err
 	}
 
-	if _, err := n.clusterProvider.GetNetwork(create.Name); err == nil {
-		return libnetwork.NetworkNameError(create.Name)
+	if nw, err := n.clusterProvider.GetNetwork(create.Name); err == nil {
```

Comment:
This check doesn't seem very useful. First, it's racy. Also, it does a prefix match, which is not what we want here.

`n.clusterProvider.CreateNetwork` will fail if the name conflicts. Could we just remove this check?